File: proselint/checks/industrial_language/airlinese.py
"""
Airlinese.

---
layout:     post
source:     Garner's Modern American Usage
source_url: http://bit.ly/1T4alrY
title:      Airlinese
date:       2014-06-10 12:31:19
categories: writing
---

Airlinese.

"""

from proselint.registry.checks import Check, types

check = Check(
    check_type=types.Existence(
        items=(
            "enplan(?:e|ed|ing|ement)",
            "deplan(?:e|ed|ing|ement)",
            "taking off momentarily",
        )
    ),
    path="industrial_language.airlinese",
    message="'{}' is airlinese.",
)

__register__ = (check,)

File: proselint-rs/crates/proselint-checks/src/industrial_language/airlinese.rs
use proselint_registry::checks::{Check, types::*, Padding};

pub const EXAMPLES_PASS: &[&str] = &[
    "Smoke phrase with nothing flagged.",
    "Did that car just hydroplane?",
    "I know how to operate a planing mill.",
];
pub const EXAMPLES_FAIL: &[&str] = &[
    "This is your captain speaking. We will be taking off momentarily.",
    "We deplaned promptly after.",
];

const CHECK: Check = Check {
	check_type: &Existence {
		items: &[
			"enplan(?:e|ed|ing|ement)",
			"deplan(?:e|ed|ing|ement)",
			"taking off momentarily",
		],
		padding: Padding::WordsInText,
		exceptions: &[],
	},
	path: "industrial_language.airlinese",
	msg: "'{}' is airlinese.",
	..Check::default()
};

pub const REGISTER: &[Check] = &[CHECK];
